.github/scripts/validate_pr_formatting.py

```diff
@@ -0,0 +1,345 @@
+#!/usr/bin/env python3
+#
+# Pull Request Formatting Validator
+#
+# This script validates basic pull request formatting requirements and can
+# optionally post validation comments.
+#
+# Validation Checks Performed:
+#
+# 1. PR Title Validation:
+#    - Ensures the PR title is not empty or whitespace-only
+#    - Failure comment: "⚠️ Pull request title cannot be empty."
+#
+# 2. PR Body Validation:
+#    - Ensures the PR body is not empty, null, or whitespace-only
+#    - Ensures the PR body meets minimum length requirement (147 characters)
+#      - The minimum length is based on the PR template with empty sections
+#    - Empty body failure comment: "⚠️ Add a meaningful pull request
+#      description using the PR template."
+#    - Short body failure comment: "⚠️ Provide a more detailed pull request
+#      description using the PR template (current: X characters)."
+#
+# 3. PR Template Line Validation:
+#    - Checks that template placeholder lines are removed from the PR
+#      description
+#       - Based on present template lines in .github/pull_request_template.md
+#    - Failure comment: "⚠️ Remove the following template lines from your
+#      PR description:" followed by list of remaining template lines
+#
+# The same comment (based on a body content hash) will not be posted multiple
+# times to avoid spamming the PR with duplicate messages.
+#
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+
+import base64
+import hashlib
+import os
+import re
+import sys
+from typing import List, Optional
+
+import GitHub
+
+
+class PRValidator:
+    """Logic for validating pull request formatting."""
+
+    VALIDATION_HEADER = "Pull Request Formatting Issues"
+    MIN_BODY_LENGTH = 147
+    TEMPLATE_LINE_PATTERN = r'^<_.*_>$'
+
+    def __init__(self, github_token: str, owner: str, repo: str,
+                 pr_number: int):
+        """Initialize the validator with GitHub context."""
+        self.github_token = github_token
+        self.owner = owner
+        self.repo = repo
+        self.pr_number = pr_number
+        self.validation_messages: List[str] = []
+        self.pr = None
+
+    def fetch_pr_data(self) -> tuple[str, str]:
+        """Fetch PR title and body."""
+        self.pr = GitHub._get_pr(
+            self.github_token, self.owner, self.repo, self.pr_number
+        )
+
+        if not self.pr:
+            print("::error title=Failed to get PR data!::"
+                  "Could not fetch PR information")
+            sys.exit(1)
+
+        title = self.pr.title or ''
+        body = self.pr.body or ''
+
+        return title, body
+
+    def fetch_template_content(self) -> Optional[str]:
+        """Fetch the PR template content if it exists."""
+        try:
+            gh = GitHub._authenticate(self.github_token)
+            repo = gh.get_repo(f"{self.owner}/{self.repo}")
+            template_file = repo.get_contents(
+                ".github/pull_request_template.md"

```

Comment:
This is a hardcoded file path in the edk2 repo.  If that file name or location changes, this script would break. This should either be configurable or be a MACRO like variable declared at top of file with comment on hard coded path.